pkg/providers/vsphere/vmlifecycle/update_status.go

```diff
@@ -1897,5 +1853,238 @@ func reconcileStatusController(
 
 	vm.Status.Hardware.Controllers = ctlStatuses
 
+	if pkgcfg.FromContext(vmCtx).Features.VMSharedDisks {
+		reconcileHardwareCondition(vmCtx, k8sClient)
+	}
+
 	return nil
 }
+
+// reconcileHardwareCondition updates the
+// VirtualMachineHardwareDeviceConfigVerified condition by verifying that the
+// VM's hardware device configuration matches the desired state specified in
+// the spec.
+func reconcileHardwareCondition(
+	vmCtx pkgctx.VirtualMachineContext,
+	k8sClient ctrlclient.Client) {
+
+	var errs []error
+
+	errs = append(errs, checkControllers(vmCtx.VM)...)
+	errs = append(errs, checkPVCVolumes(vmCtx.VM)...)
+	errs = append(errs, checkCDROMDevices(vmCtx, k8sClient)...)
+
+	if len(errs) == 0 {
+		conditions.MarkTrue(
+			vmCtx.VM,
+			vmopv1.VirtualMachineHardwareDeviceConfigVerified)
+		return
+	}
+
+	conditions.MarkFalse(
+		vmCtx.VM,
+		vmopv1.VirtualMachineHardwareDeviceConfigVerified,
+		vmopv1.VirtualMachineHardwareDeviceConfigMismatchReason,
+		"%s",
+		apierrorsutil.NewAggregate(errs).Error())
```

Comment:
I'm wondering if we should have separate conditions for controllers, volumes, cdroms (and whatever else comes along later) and then maybe roll things up to this single condition? This list could get unwieldy

Just want to have the discussion - no need to update this MR right now